.github/workflows/codeql.yml

```diff
@@ -0,0 +1,45 @@
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - master
+      - release-*
+  pull_request: {}
```

Comment:
Happy for you to optimize here, but I think only running it on merges is a perfectly acceptable solution for now :)

Reply:
I'm gonna try to see if it can be reduced to some reasonable time with reasonable effort, otherwise I agree, on merge should be good enough.

Reply:
@khos2ow gotcha 👍 I would mention here though that running it at all on every PR feels a little wasteful to me, so I wouldn't invest too much time in optimization